test/multilocale/localeTree/iterators.good

```diff
@@ -0,0 +1,12 @@
+Broadcast: 0 -> 1
+Broadcast: 0 -> 2
+Broadcast: 1 -> 3
+Broadcast: 1 -> 4
+Broadcast: 2 -> 5
+Broadcast: 2 -> 6
+Gather: 1 -> 0
+Gather: 2 -> 0
+Gather: 3 -> 1
+Gather: 4 -> 1
+Gather: 5 -> 2
+Gather: 6 -> 2
```

Comment:
I think this test should be written in such a way that we verify that the iteration order is correct.